mypy/report.py

```diff
@@ -243,10 +243,10 @@ def _write_out_report(self,
             f.write(separator + '\n')
             for row_values in rows:
                 r = ("{:>{}}" * len(widths)).format(*itertools.chain(*zip(row_values, widths)))
-                f.writelines(r + '\n')
+                f.write(r + '\n')
```

Comment:
What made this change necessary?

Reply:
That looks like it fixes a bug! writelines takes a sequence of strings, not a single string (though because a string is a sequence of strings and writelines just writes the concatenation of the items it worked :-).

Though I'm not sure how the change found this bug -- does the TextIoWriter class (or whatever is used) not have writelines?

Reply:
We now infer f as TextIOWrapper. TextIOWrapper and TextIOBase take `List[str]` for `writelines` in typeshed. Maybe that signature should be loosened though.

Reply:
Yes, I think that should be `Iterable[str]`. I just tried in 3.6 and TextIOWrapper.writelines accepts at least tuples and generator expressions.

Reply:
And it looks like the perennial "str is a iterable of str" bug is benign in this case, so there's not really any downside to the looser signature.